vhost-user-backend/Cargo.toml

```diff
@@ -8,9 +8,13 @@ repository = "https://github.com/rust-vmm/vhost"
 edition = "2021"
 license = "Apache-2.0"
 
+[package.metadata.docs.rs]
+all-features = true
+
 [features]
 xen = ["vm-memory/xen", "vhost/xen"]
 postcopy = ["vhost/postcopy", "userfaultfd"]
+DUMMY_FEATURE_disable_compile_error = ["vhost/DUMMY_FEATURE_disable_compile_error"]
```

Comment:
What about avoiding this and adding on both Cargo.toml:
```
[lints.rust]
unexpected_cfgs = { level = "warn", check-cfg = ['cfg(DUMMY_FEATURE_disable_compile_error)'] }
```

Reply:
I mean, I'd like to avoid that `cargo build --all-features` go back to work disabling the compile error.

Reply:
@roypat can we set it in the buildkite steps as env var?

Reply:
I think the best I can offer is 

```
./rust-vmm-ci/.buildkite/autogenerate_pipeline.py | sed "s/cargo doc/RUSTFLAGS='-cfg DUMMY_FEATURE_disable_compile_error' cargo doc/g"
```

in the pipeline definition instead of what we're doing right now

Reply:
yep, let's try that, plus removing the `DUMMY_FEATURE_disable_compile_error` from `[features]` and adding the `unexpected_cfgs`, right?

Reply:
@roypat yeah, let's do that for now